### PR TITLE
XChaCha20

### PR DESCRIPTION
**Pull request by tarcieri**
Support for the extended nonce variant of ChaCha20

### CODE REVIEW

**Comment by tarcieri** [comment]
Somehow this is passing the HChaCha20 test vectors, but failing the XChaCha20 vectors? I'm really confused how that's even possible.

@paragonie-scott having trouble reproducing your test vectors... I don't suppose you happen to see anything I'm doing wrong?

**Comment by tarcieri** [comment]
Aah, I think I see... the test vectors omit the first 64-bytes of the keystream (I guess that's the counter: 1)

**Comment by paragonie-scott** [comment]
I was about to say, check the block counter. :)

File: chacha20/Cargo.toml [hunk 1]
```diff
@@ -5,11 +5,12 @@ authors = ["RustCrypto Developers"]
 license = "MIT OR Apache-2.0"
 description = "Chacha20 Stream Cipher"
 repository = "https://github.com/RustCrypto/stream-ciphers"
-keywords = ["crypto", "stream-cipher", "trait"]
+keywords = ["chacha20poly1305", "crypto", "hchacha20", "stream-cipher", "xchacha20"]
```
Comment (newpavlov) [inline chacha20/Cargo.toml]: Are you sure about `chacha20poly1305` flag? Or do you intend to add it later?

Comment (tarcieri) [reply chacha20/Cargo.toml]: Good point... we should just ship a `chacha20poly1305` crate whenever https://github.com/RustCrypto/traits/pull/40 lands

**Comment by tarcieri** [comment]
@newpavlov look good now?

**Pull request merged**